snapcraft/tests/unit/project_loader/grammar/test_on_statement.py

```diff
@@ -151,7 +152,12 @@ class OnStatementGrammarTestCase(GrammarTestCase):
         }),
     ]
 
-    def test_on_statement_grammar(self):
+    @patch('platform.architecture')
+    @patch('platform.machine')
+    def test_on_statement_grammar(self, platform_machine_mock,
+                                  platform_architecture_mock):
+        platform_machine_mock.return_value = 'x86_64'
+        platform_architecture_mock.return_value = ('64bit', 'ELF')
         options = snapcraft.ProjectOptions(target_deb_arch=self.target_arch)
```

Comment:
This test case was written with the idea that it could switch what it was comparing its selectors against. I don't think it's testing the same cases, now. Is there a way to preserve that now that it's no longer based on the target arch?

Reply:
By switch you mean pass on different architectures? Generally it's more reliable to use scenarios for that rather than relying on the architecture of the test machine.
I changed it now to run all tests with both amd64 and i386 target architectures. Let me know what you think.

Reply:
Perhaps to expand a little, the tests will no longer pass on different host architectures since that's the bit we're testing now. For example `on amd64` won't trigger on anything but amd64 hosts, unlike before where only the target was relevant.

Reply:
Arg, I just realized somehow I didn't actually finish the commit... that explains the test failure.  Pushed it now...

Reply:
> For example on amd64 won't trigger on anything but amd64 hosts, unlike before where only the target was relevant.

Indeed, perhaps I wasn't clear. The tests were previously setting target arch. I'd like to see the tests unchanged except that they're now setting the host arch instead of the target. That way they're still testing the same things.

Reply:
I went back to how they were and swapped the target arch for the host arch (plus adjusting them so they pass). Also dropped the target_arch from a few tests that didn't use it at all while I was at it. But It's rather tedious to go through these so I hope we are on the same page now :-D